Sources/SwiftDocC/Converter/TopicRenderReferenceEncoder.swift

```diff
@@ -60,20 +83,32 @@ enum TopicRenderReferenceEncoder {
                     
                     let conformanceHash = Checksum.md5(of: Data(conformance.constraints.map({ $0.plainText }).joined().utf8))
                     let cacheKeyWithConformance = "\(key) : \(conformanceHash)"
-                    if let cached = referenceCache.sync({ $0[cacheKeyWithConformance] }) {
-                        value = cached
+                    if let (reference, overrides) = referenceCache.sync({ $0[cacheKeyWithConformance] }) {
+                        value = reference
+                        variantOverrides.append(contentsOf: overrides)
                     } else {
-                        value = try! encoder.encode(CodableRenderReference.init(reference))
-                        referenceCache.sync({ $0[cacheKeyWithConformance] = value })
+                        encoderUserInfoVariantOverrides.values.removeAll()
```

Comment:
Ah right- so we only want the variant overrides that are collected while encoding that single reference. Since the cache contains the reference and the overrides for that reference.

But this is definitely confusing- I'll add a comment.

Reply:
Addressed here: 5ef11e1423772d970f50e4a37d953545c66f31e6.

Thanks!